core/trino-main/src/main/java/io/trino/operator/join/SortedPositionLinks.java

```diff
@@ -109,31 +110,41 @@ public Factory build()
             ArrayPositionLinks.FactoryBuilder arrayPositionLinksFactoryBuilder = ArrayPositionLinks.builder(size);
             int[][] sortedPositionLinks = new int[size][];
 
-            for (Int2ObjectMap.Entry<IntArrayList> entry : positionLinks.int2ObjectEntrySet()) {
+            Iterator<Int2ObjectMap.Entry<IntArrayList>> iterator = positionLinks.int2ObjectEntrySet().fastIterator();
+            while (iterator.hasNext()) {
+                Int2ObjectOpenHashMap.Entry<IntArrayList> entry = iterator.next();
                 int key = entry.getIntKey();
-                IntArrayList positions = entry.getValue();
-                positions.sort(comparator);
-
-                sortedPositionLinks[key] = new int[positions.size()];
-                for (int i = 0; i < positions.size(); i++) {
-                    sortedPositionLinks[key][i] = positions.getInt(i);
-                }
-
-                // ArrayPositionsLinks.Builder::link builds position links from
-                // tail to head, so we must add them in descending order to have
-                // smallest element as a head
-                for (int i = positions.size() - 2; i >= 0; i--) {
-                    arrayPositionLinksFactoryBuilder.link(positions.getInt(i), positions.getInt(i + 1));
-                }
-
-                // add link from starting position to position links chain
-                if (!positions.isEmpty()) {
-                    arrayPositionLinksFactoryBuilder.link(key, positions.getInt(0));
+                IntArrayList positionsList = entry.getValue();
+
+                int[] positions = positionsList.toIntArray();
```

Comment:
if `it.unimi.dsi.fastutil.ints.IntArrayList#unstableSort` is used then we don't need to perform copy at all